stride/problem/domain.py

```diff
@@ -74,21 +77,26 @@ def extended_size(self):
         """
         return self.extended_limit
 
-    def resample(self, spacing, extra=None, absorbing=None):
-        if isinstance(spacing, float):
-            spacing = (spacing,)*self.dim
+    def resample(self, new_spacing, new_extra=None, new_absorbing=None):
+        '''
+        Resample space
+        '''
 
-        shape = tuple((np.array(self.size) / np.array(spacing) + 1).astype(int))
+        if isinstance(new_spacing, float):
+            new_spacing = (new_spacing,)*self.dim
 
-        if extra is None:
-            extra = tuple((np.array(self.spacing) * (np.array(self.extra) - 1) /
-                           np.array(spacing) + 1).astype(int))
+        new_shape = tuple((np.round(np.array(self.size) / np.array(new_spacing)) + 1).astype(int))
+        # TODO there is possibly a rounding error
 
-        if absorbing is None:
-            absorbing = tuple((np.array(self.spacing) * (np.array(self.absorbing) - 1) /
-                               np.array(spacing) + 1).astype(int))
+        if new_extra is None:
+            new_extra = tuple((np.round(np.array(self.spacing) * (np.array(self.extra) - 1) /
+                           np.array(new_spacing)) + 1).astype(int))
 
-        return Space(shape=shape, spacing=spacing, extra=extra, absorbing=absorbing)
+        if new_absorbing is None:
+            new_absorbing = tuple((np.round(np.array(self.spacing) * (np.array(self.absorbing) - 1) /
+                               np.array(new_spacing)) + 1).astype(int))
+
+        self._set_properties(shape=new_shape, spacing=new_spacing, extra=new_extra, absorbing=new_absorbing)
```

Comment:
like in Time, we should also reset all cached properties of the Space

Reply:
done